File: hackerrank/find-digits.rb
#!/bin/ruby

require 'json'
require 'stringio'

def findDigits(n)
    number = n
    count = 0

    while number > 0 do
        digit = number % 10
        number /= 10
        
        count += 1 if digit != 0 and n % digit == 0
    end

    return count
end

fptr = File.open(ENV['OUTPUT_PATH'], 'w')

t = gets.to_i

t.times do |t_itr|
    n = gets.to_i

    result = findDigits n

    fptr.write result
    fptr.write "\n"
end

fptr.close()

File: hackerrank/find-digits.py
#!/bin/python3

import os

def findDigits(n):
    n_str = str(n)
    count = 0

    for digit in set(n_str):
        if digit != '0' and n % int(digit) == 0:
            count += n_str.count(digit)

    return count


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        n = int(input())

        result = findDigits(n)

        fptr.write(str(result) + '\n')

    fptr.close()
